### PR TITLE
feat: add byte_buf_eq job to showcase is_modified

### PR DESCRIPTION
**Pull request by priyasiddharth**
this version seems to be *slightly* faster - 0.95s vs 1.13s - but need more data.

### CODE REVIEW

File: seahorn/jobs2/byte_buf_eq2/aws_byte_buf_eq_harness2.c [hunk 1]
```diff
@@ -0,0 +1,55 @@
+/*
+ *
+ */
+
+#include <aws/common/byte_buf.h>
+#include <byte_buf_helper.h>
+#include <seahorn/seahorn.h>
+#include <utils.h>
+
+extern bool sea_is_modified(char *);
```
Comment (agurfinkel) [inline seahorn/jobs2/byte_buf_eq2/aws_byte_buf_eq_harness2.c:10]: these should eventually be moved to seahorn.h

File: seahorn/jobs2/byte_buf_eq2/aws_byte_buf_eq_harness2.c [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  /* assumptions */
+  assume(aws_byte_buf_is_valid(&lhs));
+  if (nd_bool()) {
+    rhs = lhs;
+  } else {
+    assume(aws_byte_buf_is_valid(&rhs));
+  }
+
+  sea_tracking_on();
+  sea_reset_modified((char *)&lhs);
```
Comment (agurfinkel) [inline seahorn/jobs2/byte_buf_eq2/aws_byte_buf_eq_harness2.c:31]: explicit reset is required? I was assuming that modified flag is 0 until tracking is enabled

**Review by agurfinkel — APPROVED** [review]

File: seahorn/jobs2/byte_buf_eq2/aws_byte_buf_eq_harness2.c (hunk 1 above)
Comment (priyasiddharth) [reply seahorn/jobs2/byte_buf_eq2/aws_byte_buf_eq_harness2.c]: moved to seahorn.h ... will send another PR in seahorn project.

File: seahorn/jobs2/byte_buf_eq2/aws_byte_buf_eq_harness2.c (hunk 2 above)
Comment (priyasiddharth) [reply seahorn/jobs2/byte_buf_eq2/aws_byte_buf_eq_harness2.c]: no not required... removed...

**Pull request merged**